bot/exts/moderation/infraction/_utils.py

```diff
@@ -188,6 +190,40 @@ async def notify_infraction(
     return await send_private_embed(user, embed)
 
 
+async def notify_infraction(
+        infraction: Infraction,
+        user: t.Optional[UserSnowflake] = None,
+        reason: t.Optional[str] = None
+) -> bool:
+    """
+    DM a user about their new infraction and return True if the DM is successful.
+
+    `user` and `reason` can be used to override what is in `infraction`. Otherwise, this data will
+    be retrieved from `infraction`.
+
+    Also return False if the user needs to be fetched but fails to be fetched.
+    """
+    if user is None:
+        user = discord.Object(infraction["user"])
+
+    # Sometimes user is a discord.Object; make it a proper user.
```

Comment:
Is there a case, other than showcased above when `user` is not passed, when this is called with a `discord.Object` instance?

Because if there's no other case, the if-statement and this fetch can be combined.

Reply:
Yes, there is. Besides, if the fetch fails, the except block needs to be able to access the `id` attribute.